salt/_modules/metalk8s_kubernetes.py

```diff
@@ -2373,3 +2405,139 @@ def node_uncordon(node_name, **kwargs):
         salt '*' kubernetes.node_uncordon  node_name="minikube"
     '''
     return node_set_unschedulable(node_name, False, **kwargs)
+
+
+def get_crd_plural_from_kind(kind, **kwargs):
+    plural = None
+
+    crds = customresourcedefinition(**kwargs)
+
+    for crd in crds.get('items', []):
+        crd_names = crd.get('spec', {}).get('names', {})
```

Comment:
For now it's use only once, but if this pattern (multiple `get`) becomes more widespread, we will want to use a recursive `get` to make the code easier to read, e.g.:

```python
def get_rec(d, keys, sep='.', default=None):
    try:
        return functools.reduce(operator.getitem, keys.split(sep), d)
    except KeyError:
        return default

def get_crd_plural_from_kind(kind, **kwargs):
    for crd in customresourcedefinition(**kwargs).get('items', []):
        if get_rec(crd, 'spec.names.kind') == kind:
            return get_rec(crd, 'spec.names.plural')
```

Reply:
There might be one in Salt already, since the `__salt__` "dict" uses the same kind of magic for its `__getitem__` (e.g. `__salt__["network.ip_addr"]`)

Reply:
Would be nice if they expose it, but maybe it's part of their internals.